File: MD_sample/Cpp_OOP/verlet.cxx
#include "md.hxx"

double dnint(double x);

/* First step of velocity verlet */
void MD::vverlet_1(double dt)
{
  unsigned i, k;
  for (i=0; i<Npt; i++) {
    for (k=0; k<3; k++) {
      q[i].xx[k] += dt * q[i].vv[k] + 0.5*dt*dt*q[i].ff[k]/q[i].xm;
      q[i].xx[k] -= box * dnint(q[i].xx[k]/box);
      q[i].vv[i] += 0.5*dt*q[i].ff[k]/q[i].xm;
    }
  }
}

/* Second step of velocity verlet */
void MD::vverlet_2(double dt)
{
  unsigned i, k;
  for (i=0; i<Npt; i++) {
    for (k=0; k<3; k++) {
      q[i].vv[i] += 0.5*dt*q[i].ff[k]/q[i].xm;
    }
  }
}

File: MD_sample/F90/verlet.f90
!
! 1st step of Velocity Verlet time integration
!############################################################
subroutine verlet_1(dt)
  use md
  implicit none
  integer(di):: i
  real   (dp):: dt
  !
  do i=1_di, Npt
     q(i)%xx(:) = q(i)%xx(:) + dt*q(i)%vv(:) + 0.5_dp*dt*dt*q(i)%ff(:)/q(i)%xm
     q(i)%xx(:) = q(i)%xx(:) - l_box*dnint(q(i)%xx(:)/l_box)
     q(i)%vv(:) = q(i)%vv(:) + 0.5_dp*dt*q(i)%ff(:)/q(i)%xm
  end do
  return
end subroutine verlet_1
!
! 2nd step of Velocity Verlet time integration
!############################################################
subroutine verlet_2(dt)
  use md
  implicit none
  integer(di):: i
  real   (dp):: dt
  !
  do i=1_di, Npt
     q(i)%vv(:) = q(i)%vv(:) + 0.5_dp*dt*q(i)%ff(:)/q(i)%xm
  end do
  return
end subroutine verlet_2

